src/components/EventRegistration.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import RegistrationCollection from './RegistrationCollection';
+
+class EventRegistration extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      collections: this.props.formGroups.reduce((acc, reg) => {
+        acc[reg.id] = [];
+        return acc;
+      }, {})
+    };
+    this.updateCollection = this.updateCollection.bind(this);
+    this.submit = this.submit.bind(this);
+    this.mergeDataAndFormGroups = this.mergeDataAndFormGroups.bind(this);
+  }
+
+  /*
+   * Is called when parent updates props, i.e. when data is updated from the outside
+   */
+  componentWillReceiveProps(nextProps) {
+    if (nextProps && nextProps.data) {
+      this.setState({ collections: this.mergeDataAndFormGroups(nextProps.data) });
+    }
+  }
+
+  mergeDataAndFormGroups(data) {
+    const result = {...data};
+    // add formGroups that are not yet present in data
+    this.props.formGroups.forEach((fg, index) => {
+      if (result[fg.id] === undefined) {
+        result[fg.id] = [];
+      }
+    });
+    return result;
+  }
+
+  collect = (data, id) => {
+    console.log('EventRegistration.collect data', data);
+    console.log('- id', id);
+    this.setState({
+      collect: {
+        ...this.state.collect,
+        [id]: data,
+      },
+    });
+  };
+
+  updateCollection(collectionId, data) {
+    const collections = {...this.state.collections};
+    collections[collectionId] = data;
+    this.setState({collections});
+    if (this.props.onChange) {
+      // TODO: Find out why this.state.collections is not updated here
+      this.props.onChange(collections);
```

Comment:
Eg skjønar ikkje korfor `this.props.onChange(this.state.collections)` ikkje gjorde same nytten her. State var ikkje oppdatert med siste endring frå addRow.